google-ads-examples/src/main/java/com/google/ads/googleads/examples/hotelads/AddHotelCallout.java

```diff
@@ -0,0 +1,231 @@
+         * @param customerId the client customer ID.
+         * @param campaignID the campaign ID.
+         * @param calloutText the text for the callout.
+         * @param languageCode the language code for the callout (IETF BCP 47 compliant).
+         * @return resource name of the newly created extension feed item.
+         * @throws GoogleAdsException if an API request failed with one or more service errors.
+         */
+        private String addExtensionFeedItem(
+            GoogleAdsClient googleAdsClient, 
+            long customerId, 
+            String calloutText, 
+            String languageCode) {
+                // Create the callout with text and language of choice
+                HotelCalloutFeedItem hotelCallout = HotelCalloutFeedItem.newBuilder()
+                    .setText(StringValue.of(calloutText))
+                    .setLanguageCode(StringValue.of(languageCode))
+                    .build();
+                
+                // Attach callout to a FeedItem
+                ExtensionFeedItem feedItem = ExtensionFeedItem.newBuilder()
+                    .setHotelCalloutFeedItem(hotelCallout)
+                    .build();
+                
+                // Create the feed item operation
+                ExtensionFeedItemOperation feedItemOperation = ExtensionFeedItemOperation.newBuilder()
+                    .setCreate(feedItem)
+                    .build();
+                
+                // Issues the create request to create the feed item
+                try (ExtensionFeedItemServiceClient extensionFeedItemServiceClient = 
+                        googleAdsClient.getLatestVersion().createExtensionFeedItemServiceClient()) {
+                            MutateExtensionFeedItemsResponse response = extensionFeedItemServiceClient
+                                .mutateExtensionFeedItems(Long.toString(customerId), ImmutableList.of(feedItemOperation));
+                            String extensionFeedItemResourceName = response.getResults(0).getResourceName();
+                            System.out.printf(
+                                "Added an extension feed item with resource name: '%s'%n", 
+                                extensionFeedItemResourceName
+                            );
+                            return extensionFeedItemResourceName;
+                }
+        }
+
+        /**
+         * Adds extension feed item to the Campaign.
+         *
+         * @param googleAdsClient the Google Ads API client.
+         * @param customerId the client customer ID.
+         * @param campaignID the campaign ID.
+         * @param extensionFeedItemResourceName the resource name for the extension feed item
+         * @return resource name of the newly created extension feed item.
+         * @throws GoogleAdsException if an API request failed with one or more service errors.
+         */
+        private String addExtensionToCampaign(
```

Comment:
Do you also want to add examples for adding these to AdGroups and account level?

Reply:
@nwbirnie is that preferred?  Only differences I see are which `ExtensionSetting` object gets used and that when using `CustomerExtensionSetting` you don't need to attach a `resource_name` to it.

Reply:
There's not really a single answer to this. In general it would make sense to cover say 80% of use-cases with a code example. So if you think most folks will set these once at the campaign level, this is fine.

Reply:
It is easy enough to add, will do one for AdGroup  and Account too.